services/core/shared/core/compat/sdk_v4/transactions.js

```diff
@@ -95,15 +96,15 @@ const getTransactions = async params => {
 	// TODO: Add search by message
 
 	const resultSet = await transactionIdx.find(params);
-
+	const [{ count }] = await transactionIdx.count();
 	if (resultSet.length > 0) {
 		const trxIds = resultSet.map(row => row.id);
 		transactions.data = await getTransactionByIds(trxIds);
 	}
 
-	transactions.meta.total = transactions.meta.count;
 	transactions.meta.count = transactions.data.length;
-	transactions.meta.offset = params.offset || 0;
+	transactions.meta.total = count;
```

Comment:
Assigning the absolute `count` value as `total` might not be the correct approach. If someone queries `/api/v1/transactions?type=10`, then not all the transactions in the DB would be fitting this user request. Hence, we'd be presenting the user an intuition that a lot more transactions of type `10` exist than actual.

Please consider performing a `COUNT('id')` for the request parameters that the user sends and returning that value. We can discuss this further offline.

Reply:
Implemented the logic for it.